recipes/openai_textcat_multilabel.py

```diff
@@ -0,0 +1,527 @@
+    prompt_path=("Path to jinja2 prompt template", "option", "p", Path),
+    max_examples=("Max examples to include in prompt", "option", "n", int),
+    exclusive_classes=("Make the classification task exclusive", "flag", "E", bool),
+    verbose=("Print extra information to terminal", "flag", "v", bool),
+    # fmt: on
+)
+def textcat_multilabel_openai_correct(
+    dataset: str,
+    input_path: Path,
+    labels: List[str],
+    lang: str = "en",
+    model: str = "text-davinci-003",
+    batch_size: int = 10,
+    segment: bool = False,
+    examples_path: Optional[Path] = None,
+    prompt_path: Path = DEFAULT_PROMPT_PATH,
+    max_examples: int = 2,
+    exclusive_classes: bool = False,
+    verbose: bool = False,
+):
+    """Get bulk TextCat suggestions from an OpenAI API, using zero-shot or
+    few-shot learning.  The results can then be corrected using the
+    `textcat.manual` recipe.
+
+    Here, we use ChatGPT to suggest if a particular text talks about a recipe or
+    not. You can use the parameter `--chatgpt-bias` (float, 0 to 1 inclusive) to
+    set how much we prefer seeing examples that ChatGPT thinks as recipes.
+    """
+
+    api_key, api_org = _get_api_credentials(model)
+    examples = _read_prompt_examples(examples_path)
+    if labels is None:
+        msg.fail("textcat.teach requires at least one --label", exits=1)
+    nlp = spacy.blank(lang)
+
+    if segment:
+        nlp.add_pipe("sentencizer")
+
+    openai = OpenAISuggester(
+        openai_model=model,
+        labels=labels,
+        max_examples=max_examples,
+        prompt_template=_load_template(prompt_path),
+        exclusive_classes=exclusive_classes,
+        verbose=verbose,
+        segment=segment,
+        openai_api_key=api_key,
+        openai_api_org=api_org,
+        openai_timeout_s=120,
+        openai_n=10,
+    )
+
+    for eg in examples:
+        openai.add_example(eg)
+
+    stream = list(srsly.read_jsonl(input_path))
+    stream = openai(tqdm.tqdm(stream), batch_size=batch_size, nlp=nlp)
+
+    return {
+        "dataset": dataset,
+        "view_id": "blocks",
+        "stream": stream,
+        "update": openai.update,
+        "config": {
+            "labels": openai.labels,
+            "batch_size": batch_size,
+            "exclude_by": "input",
+            "choice_style": "single" if exclusive_classes else "multiple",
+            "blocks": [
+                {"view_id": "choice"},
+                {"view_id": "html", "html_template": HTML_TEMPLATE},
+            ],
+            "show_flag": True,
+            "global_css": CSS_FILE_PATH.read_text(),
+        },
+    }
+
+
+@prodigy.recipe(
+    "textcat_multilabel.openai.fetch",
+    input_path=("Path to jsonl data to annotate", "positional", None, Path),
+    output_path=("Path to save the output", "positional", None, Path),
+    labels=("Labels (comma delimited)", "positional", None, lambda s: s.split(",")),
```

Comment:
Total nitpick, but shouldn't `labels` be an option `--labels`? Mainly to keep them consistent with the [Prodigy recipes](https://prodi.gy/docs/recipes/#textcat-correct)?